Register-Share-System/src/Server/UDPServer.java

```diff
@@ -50,29 +59,73 @@ public void run() {
     	while(true) {	//loop for receiving/parsing/handling incoming data
 	     	
     		dpReceive = new DatagramPacket(inputBuffer,inputBuffer.length);
-    			
-    		System.out.println("\nwaiting to recieve data");
+    		
     	    try { serverSocket.receive(dpReceive); }			//wait till data is received
     	    catch (IOException e) { e.printStackTrace(); System.out.println("socketException while recieving data");}
     	    
+    	    
+    	    
     	    //----------------------------
     	    // server input parser/handler
     	    //----------------------------
     	    
+    	    /*	0: test case
+    	     * 	1: registration request
+    	     * 	100: server sync
+    	     * 	101: server sync confirmation
+    	     * 	101: server switch
+    	     * 	102: server switch
+    	     */
+    	    
+    	    
+    	    if(isServing)				//socket input handlers which run while serving
     	    switch(inputBuffer[0]) {
     	    	
     	    case 0:	// a test case, print message to console, and respond with 'message received'
     	    	server_app.display("data recieved from client: "+parseString(inputBuffer,1));	//convert data to string, then send to main for displaying
-    	    	sendString("message recieved", 0, dpReceive.getAddress());					//send response
+    	    	sendString("message recieved", 0, dpReceive.getAddress(), dpReceive.getPort());	//send response
     	    	break;
     	    	
     	    case 1: //registration request
     	    	
     	    	
-    	    default:
-    	    	server_app.display("invalid operation recieved, initial byte out of range");
     	    }
     	    
+    	    switch(inputBuffer[0]) {	//socket input handler that runs even when not serving
+    	    
+    	    case 100:	//server sync: 
+    	    	
+    	    	//save info for other server
+    	    	otherServerIP = dpReceive.getAddress();
+    	    	otherServerPort = dpReceive.getPort();
+    	    	
+    	    	if(Objects.equals(parseString(inputBuffer,1),"s1")) {	//we are server 1
+        	    	dualServerSync = true;	//set server as sync'd
+        	    	isServing = true;		//server 1 serves first
+        	    	System.out.println("this server is serving");
+        	    	sendServer("s2",100);	//respond to sync other server
+    	    	} else {	//we are server 2
+    	    		dualServerSync = true;	//set server as sync'd but do not set isServing
+    	    		sendServer("s",101);	//respond with sync confirmation to start server1's timer
+    	    		System.out.println("this server is not serving");
+    	    		serverSwitchTimer = new Timer();		//initialize server 2's timer
+        	    	serverSwitchExec = new TimerExec();		//initialize server2's timerTask
+    	    	}
+    	    	break;
+    	    	
+    	    case 101: //server sync confirmation
+    	    	
+    	    	//if we're here, then both servers are now sync'd and this is server 1
+    	    	//start serverSwitchTimer
+    	    	startTimer();
+    	    	break;
+    	    
+    	    case 102: //server switch
```

Comment:
I think that the serving server is incharge of when a switch is supposed to happen